src/utils/load_data.jl

```diff
@@ -1,83 +1,313 @@
+- Adding the information in the `*.csv` file at `data_path` to the data dictionary
+The `*.csv` files shall have the following structure and must have the same length:
+|`Timestamp` |`Year`  |[column names...]|
+|----------|------|-----------------|
+|[iterator]|[year]|[values]         |
+The first column should be called `Timestamp` if it contains a time iterator
+The other columns can specify the single timeseries like specific geolocation.
+for regions:
 - `"GER_1"`: Germany 1 node
 - `"GER_18"`: Germany 18 nodes
 - `"CA_1"`: California 1 node
 - `"CA_14"`: California 14 nodes
 - `"TX_1"`: Texas 1 node
 """
-function load_timeseries_data( application::String,
-                              region::String;
-                              K=365::Int,
-                              T=24::Int
-                              )
-  dt = Dict{String,Array}()
-  num=0
-  data_path=normpath(joinpath(dirname(@__FILE__),"..","..","data",application,region,"TS"))
-  for fulldataname in readdir(data_path)
-      dataname=split(fulldataname,".")[1]
-      data_df=CSV.read(joinpath(data_path,fulldataname);allowmissing=:none)
-      for column in eachcol(data_df, true)
-          if findall([:Timestamp,:time,:Time,:Zeit].==(column[1]))==[]
-              dt[dataname*"-"*string(column[1])]=Float64.(column[2])
-              newnum=length(column[2])
-              if newnum!=num && num!=0
-                  @error("The TimeSeries have different lengths!")
-              else
-                  num=newnum
-              end
-          end
-      end
-  end
-  data_full =  FullInputData(region, num, dt)
-  data_reshape =  ClustData(data_full,K,T)
-  return data_reshape, data_full
-end #load_pricedata
+function load_timeseries_data_provided(region::String="GER_1";
+                              T::Int64=24,
+                              years::Array{Int64,1}=[2016],
+                              att::Array{String,1}=Array{String,1}())
+    # Generate the data path based on application and region
+    data_path=normpath(joinpath(dirname(@__FILE__),"..","..","data",region,"TS"))
+    return load_timeseries_data(data_path; region=region, T=T, years=years, att=att)
+end
 
 """
-    load_cep_data(region::String)
-Loading from .csv files in a the folder ../ClustForOpt/data/CEP/{region}/
-Follow instructions for the CSV-Files:
-- `nodes`:       `nodes x region, infrastruct, capacity-of-different-tech... in MW_el`
-- `var_costs`:     `tech x [USD for fossils: in USD/MWh_el, CO2 in kg-CO₂-eq./MWh_el]` # Variable costs per year
-- `fix_costs`:     `tech x [USD in USD/MW_el, CO2 in kg-CO₂-eq./MW_el]` # Fixed costs per year
-- `cap_costs`:     `tech x [USD in USD/MW_el, CO2 in kg-CO₂-eq./MW_el]` # Entire (NOT annulized) Costs per Investment in technology
-- `techs`:        `tech x [categ,sector,lifetime in years,effic in %,fuel]`
-- `lines`:       `lines x [node_start,node_end,reactance,resistance,power,voltage,circuits,length]`
+    load_cep_data_techs(data_path::String)
```

Comment:
Entire new structure:
- each former table is a OptVariable now: that means that they have sets and can be indexed like Arrays
- each nodes, costs, techs and lines has it's own load-function: Within the function the necessary values are extracted and the enry specific struct is filled: the value at each OptVariable entry is specific to be either
- Number
- OptDataCEPNode
- OptDataCEPLine
- OptDataCEPTech

the single OptDataCEP... hold specific information that needs to be provided. The load functions shall be expanded in the future to handle, if certain values are not provided -> automatically assign 0 or something similar and warn user about certain missing values in a nice way...

Reply:
Awesome! Do you have examples of that in the workflow example?

Reply:
Well, it's called when loading_cep_data. The user just needs to prepare the .csv files, which is explained in detail (https://youngfaithful.github.io/CEP.jl/dev/csv_structure/)